### PR TITLE
chore: remove vendor directory

### PR DESCRIPTION
**Pull request by tobiasgiese**
Switched to Go modules for dependency management. This simplifies our workflow and reduces repo size.

### CODE REVIEW

**Review by adrianchiris — APPROVED** [review]
if CI is green im on board with that

**Comment by bn222** [comment]
if CI is green, I'm also on board with this

**Comment by tobiasgiese** [comment]
Having issues (locally) with `controller-gen`

```
go: gopkg.in/check.v1@v0.0.0-20161208181325-20d25e280405: missing go.sum entry for go.mod file
```

Added `gopkg.in/check.v1` to `hack/tools.go` already but with no lock. Also the exact version is in the go.sum already.

**Comment by tobiasgiese** [comment]
@adrianchiris @bn222 FYI - had to add `export GOPATH=$(shell go env GOPATH)` to the Makefile, otherwise the GH runners downloaded all the deps to the repos `go/`.
Also removed the unnecessary vendor tests in https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/739/commits/900812436dd1895883b6fe996c32609d406786c9

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -58,22 +58,6 @@ jobs:
     - name: test bindata/scripts
       run: make test-bindata-scripts
 
-  modules:
```
Comment (adrianchiris) [inline .github/workflows/test.yml:61]: switch the check to:

1. run go mod tidy
2. check git diff 

WDYT?

Comment (tobiasgiese) [reply .github/workflows/test.yml]: Good idea. Implemented and pushed

**Comment by tobiasgiese** [comment]
I think the issue is that the GH Actions runner has no *real* home directory. Also `go env GOPATH` is empty. Maybe the issue are the `srirov` runners?

**Comment by tobiasgiese** [comment]
```
❯ go env GOPATH
/Users/tgiese/go/
❯ echo $GOPATH
/Users/tgiese/go/
❯ unset GOPATH
❯ go env GOPATH
/Users/tgiese/go
❯ export HOME=
❯ go env GOPATH

❯
```

Seems like the issue is really the empty `$HOME`

File: Makefile [hunk 2]
```diff
@@ -27,7 +27,7 @@ WEBHOOK_IMAGE_TAG?=$(IMAGE_REPO)/$(APP_NAME)-webhook:latest
 MAIN_PKG=cmd/manager/main.go
 export NAMESPACE?=openshift-sriov-network-operator
 export WATCH_NAMESPACE?=openshift-sriov-network-operator
-export GOFLAGS+=-mod=vendor
+export GOPATH=$(shell go env GOPATH)
```
Comment (zeeke) [inline Makefile:30]: can you change this to:
```
export HOME?=$(PWD)

```

also, can you split this commit that files deletions are separated to other file changes?  It It makes very hard to browse changes.

Comment (tobiasgiese) [reply Makefile:30]: Ah, good idea! 👍🏻 
Sure, let me split it

Comment (tobiasgiese) [reply Makefile:30]: The initial idea why I changed this line in the removal commit was that each commit is working. But I can separate them for sure 👍🏻

Comment (zeeke) [reply Makefile]: Yeah, that's a good principle. But in GH it makes reviews pretty impossible :sweat_smile:

**Comment by tobiasgiese** [comment]
/test-e2e-nvidia-all
was low on disk space, added 50G

**Comment by tobiasgiese** [comment]
/test-e2e-nvidia-all

all tests are running now 👍🏻

**Review by adrianchiris — APPROVED** [review]

**Comment by adrianchiris** [comment]
thanks for working on this one @tobiasgiese  !! 
@zeeke PTAL, lets merge this one if you are happy with it.

**Review by Eoghan1232 — APPROVED** [review]

**Comment by adrianchiris** [comment]
punching it ! 🚀

**Pull request merged**